src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/KeyStatisticService.cs

```diff
@@ -0,0 +1,192 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using GovUk.Education.ExploreEducationStatistics.Admin.Requests;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.ManageContent;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.ViewModels;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services.ManageContent;
+
+public class KeyStatisticService : IKeyStatisticService
+{
+    private readonly ContentDbContext _context;
+    private readonly IPersistenceHelper<ContentDbContext> _persistenceHelper;
+    private readonly IDataBlockService _dataBlockService;
+    private readonly IUserService _userService;
+    private readonly IMapper _mapper;
+
+    public KeyStatisticService(
+        ContentDbContext context,
+        IPersistenceHelper<ContentDbContext> persistenceHelper,
+        IDataBlockService dataBlockService,
+        IUserService userService,
+        IMapper mapper)
+    {
+        _context = context;
+        _persistenceHelper = persistenceHelper;
+        _dataBlockService = dataBlockService;
+        _userService = userService;
+        _mapper = mapper;
+    }
+
+    public async Task<Either<ActionResult, KeyStatisticDataBlockViewModel>> CreateKeyStatisticDataBlock(
+        Guid releaseId,
+        KeyStatisticDataBlockCreateRequest request)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async _ =>
+                await _persistenceHelper.CheckEntityExists<DataBlock>(request.DataBlockId))
+            .OnSuccessVoid(async dataBlock =>
```

Comment:
Hey Mark, I'm wondering if there's a way to do this without having to unpack the Either explicitly, to be more in line with our usual chaining?

Something like:

```
.OnSuccess(_ =>_dataBlockService.GetAvailableDataBlocks(releaseId))
.OnSuccessVoid(availableDataBlocks => {
    if (!availableDataBlocks.Any(available => available.Id == dataBlock.Id))
    {
          throw new ArgumentException("Data block should be available");
    }
})
.OnSuccess(async _ => .......)
```

or even:

```
.OnSuccess(_ =>_dataBlockService.GetAvailableDataBlocks(releaseId))
.OnSuccess(availableDataBlocks => 
  availableDataBlocks.Any(available => available.Id == dataBlock.Id)) ? availableDataBlocks : ValidationUtils.ValidationActionResult(DataBlockShouldBeAvailable))
.OnSuccess(async _ => .......)
```